clang/lib/Sema/SemaOpenMP.cpp

```diff
@@ -7132,7 +7118,7 @@ ExprResult SemaOpenMP::ActOnOpenMPCall(ExprResult Call, Scope *Scope,
   if (!CalleeFnDecl)
     return Call;
 
-  if (getLangOpts().OpenMP >= 51 && getLangOpts().OpenMP < 60 &&
+  if (getLangOpts().OpenMP >= 50 && getLangOpts().OpenMP <= 60 &&
```

Comment:
OpenMP 5.0 and 6.0 also have the same restrictions as 5.1 and 5.2 w.r.t. OpenMP runtime APIs (e.g., `omp_get_thread_num()`) inside a construct specified with the `order(concurrent)` clause.